src/modules/displayMeals.js

```diff
@@ -0,0 +1,270 @@
+  }
+};
+
+const createCommentsSection = (comments) => {
+  const commentsSection = document.createElement('div');
+  commentsSection.classList.add('comments-section');
+
+  const commentCount = comments.length;
+
+  if (commentCount > 0) {
+    const commentsList = document.createElement('ul');
+    commentsList.classList.add('comments-list');
+
+    const reversedComments = comments.slice().reverse();
+
+    const commentCounter = document.createElement('p');
+    commentCounter.classList.add('comment-counter');
+    commentCounter.textContent = `Number of Comments : ${commentCount}`;
+
+    commentsSection.appendChild(commentCounter);
+
+    updateCommentCounter(commentCount);
+
+    reversedComments.forEach((comment) => {
+      const commentItem = document.createElement('li');
+      commentItem.classList.add('comment');
+
+      const commentContainer = document.createElement('div');
+      commentContainer.classList.add('comment-container');
+
+      const commentMeta = document.createElement('div');
+      commentMeta.classList.add('comment-meta');
+
+      const commentAuthor = document.createElement('span');
+      commentAuthor.classList.add('comment-author');
+      commentAuthor.textContent = ` by ${comment.username}`;
+
+      const commentDateTime = document.createElement('span');
+      commentDateTime.classList.add('comment-datetime');
+      commentDateTime.textContent = `${comment.creation_date}`;
+
+      commentMeta.appendChild(commentDateTime);
+      commentMeta.appendChild(commentAuthor);
+
+      const commentText = document.createElement('span');
+      commentText.classList.add('comment-text');
+      commentText.textContent = comment.comment;
+
+      commentContainer.appendChild(commentMeta);
+      commentContainer.appendChild(commentText);
+
+      commentItem.appendChild(commentContainer);
+
+      commentsList.appendChild(commentItem);
+    });
+
+    commentsSection.appendChild(commentsList);
+  } else {
+    const noCommentsMessage = document.createElement('p');
+    noCommentsMessage.textContent = 'Be the first one Leave a Comment';
+    commentsSection.appendChild(noCommentsMessage);
+  }
+
+  return commentsSection;
+};
+
+const updateCommentsSection = (newCommentsSection) => {
+  const existingCommentsSection = document.querySelector('.comments-section');
+  if (existingCommentsSection) {
+    existingCommentsSection.replaceWith(newCommentsSection);
+  }
+};
+
+const fetchCommentsForMeal = async (mealId) => {
+  const apiUrl = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/VT6MgkCI6GHRm8y0ZkMt/comments?item_id=${mealId}`;
+
+  try {
+    const response = await fetch(apiUrl);
+
+    if (!response.ok) {
+      console.error('Error response:', response.status, await response.text());
+
+      return [];
+    }
+
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error('Error fetching comments:', error);
+    throw error;
+  }
+};
```

Comment:
- [ ] Ouch! you're doing so well however as part of the project requirements the functionality for the comments counter should be implemented as a separate module, so I'd advise you to create a separate module for fetching the number of comments as this should be covered in the unit tests.

Reply:
Thanks a lot !

Reply:
Fixed.